base/regex.jl

```diff
@@ -285,6 +285,11 @@ iterate(m::RegexMatch, args...) = iterate(m.captures, args...)
 length(m::RegexMatch) = length(m.captures)
 eltype(m::RegexMatch) = eltype(m.captures)
 
+NamedTuple(m::RegexMatch) = NamedTuple{Symbol.(Tuple(keys(m)))}(values(m))
+Dict(m::RegexMatch) = Dict(pairs(m))
+Dict{String}(m::RegexMatch) = Dict(string.(keys(m)) .=> values(m))
+Dict{Symbol}(m::RegexMatch) = Dict(Symbol.(keys(m)) .=> values(m))
```

Comment:
This one seems odd. On the one hand, it is somewhat descriptive of what the Dict should look like in structure without excess information. On the other hand, there do not exist any other Dict constructors like this one. I think we need to consider adding the more general cases of `Dict{K}(pairs) where {K}` and `(Dict{<:Any,V}(pairs) where V` before we add this one

Reply:
features generally should be either cohesive or standalone, and this currently wouldn't be either